pkg/snmp/traps/traps_db.go

```diff
@@ -10,6 +10,11 @@ type VariableMetadata struct {
 	Name        string         `yaml:"name" json:"name"`
 	Description string         `yaml:"descr" json:"descr"`
 	Enumeration map[int]string `yaml:"enum" json:"enum"`
+	isNode      bool
+	// In theory, variables should always be leaves of the OID tree as intermediate nodes do not contain data.
+	// This isn't true in practice (see 1.3.6.1.4.1.4962.2.1.6.3).
+	// Variables are resolved by 'climbing' up the OID tree until finding a match, but variables that are known to be nodes
+	// should never be used for resolving.
```

Comment:
If by isNode we mean an OID that is not a leaf, I think the opposite `isLeaf` might be more readable. 

Semantically, a leaf is also a node, so `isNode` can create confusion IMHO.

Reply:
That's a good point, unfortunately I also wanted a field that would default to false as we expect most of OIDs in this struct to be regular leaves.

What about `isIntermediateNode` ? Or `isNonLeafNode` ?

Reply:
Updated with `isIntermediateNode` instead, let me know if you'd like something else